src/ruby/ext/grpc/rb_compression_options.c

```diff
@@ -0,0 +1,351 @@
+
+  grpc_compression_options_disable_algorithm(wrapper->wrapped,
+                                             compression_algorithm);
+
+  return Qnil;
+}
+
+/* Provides a bitset as a ruby number that is suitable to pass to
+ * the GRPC core as a channel argument to enable compression algorithms. */
+VALUE grpc_rb_compression_options_get_enabled_algorithms_bitset(VALUE self) {
+  grpc_rb_compression_options *wrapper = NULL;
+
+  TypedData_Get_Struct(self, grpc_rb_compression_options,
+                       &grpc_rb_compression_options_data_type, wrapper);
+  return INT2NUM((int)wrapper->wrapped->enabled_algorithms_bitset);
+}
+
+void grpc_rb_compression_options_set_default_level_helper(
+    grpc_compression_options *compression_options,
+    grpc_compression_level level) {
+  compression_options->default_level.is_set |= 1;
+  compression_options->default_level.level = level;
+}
+
+/* Sets the default compression level, given the name of a compression level.
+ * Throws an error if no algorithm matched. */
+VALUE grpc_rb_compression_options_set_default_level(VALUE self,
+                                                    VALUE new_level) {
+  char *level_name = NULL;
+  grpc_rb_compression_options *wrapper = NULL;
+  long name_len = 0;
+  VALUE ruby_str = Qnil;
+
+  TypedData_Get_Struct(self, grpc_rb_compression_options,
+                       &grpc_rb_compression_options_data_type, wrapper);
+
+  /* Take both string and symbol parameters */
+  ruby_str = rb_funcall(new_level, rb_intern("to_s"), 0);
+
+  level_name = RSTRING_PTR(ruby_str);
+  name_len = RSTRING_LEN(ruby_str);
+
+  /* Check the compression level of the name passed in, and see which macro
+   * from the GRPC core header files match. */
+  if (strncmp(level_name, "none", name_len) == 0) {
+    grpc_rb_compression_options_set_default_level_helper(
+        wrapper->wrapped, GRPC_COMPRESS_LEVEL_NONE);
+  } else if (strncmp(level_name, "low", name_len) == 0) {
+    grpc_rb_compression_options_set_default_level_helper(
+        wrapper->wrapped, GRPC_COMPRESS_LEVEL_LOW);
+  } else if (strncmp(level_name, "medium", name_len) == 0) {
+    grpc_rb_compression_options_set_default_level_helper(
+        wrapper->wrapped, GRPC_COMPRESS_LEVEL_MED);
+  } else if (strncmp(level_name, "high", name_len) == 0) {
+    grpc_rb_compression_options_set_default_level_helper(
+        wrapper->wrapped, GRPC_COMPRESS_LEVEL_HIGH);
+  } else {
+    rb_raise(rb_eNameError,
+             "Invalid compression level name. Supported levels: none, low, "
+             "medium, high");
+  }
+
+  return Qnil;
+}
+
+/* Gets the internal value of a compression algorithm suitable as the value
+ * in a GRPC core channel arguments hash.
+ * Raises an error if the name of the algorithm passed in is invalid. */
+void grpc_rb_compression_options_get_internal_value_of_algorithm(
+    VALUE algorithm_name, grpc_compression_algorithm *compression_algorithm) {
```

Comment:
Why not just have this function return the `grpc_compression_algorithm`. You just have one out parameter, and you don't even have a return value.